src/middlewares/error.middleware.ts

```diff
@@ -1,32 +1,78 @@
+import { DEV_ENV } from '@config';
+import { ErrorsMessages } from '@constants/errorMessages';
+import { HttpStatusCode } from '@constants/httpStatusCode';
+import { ValidationError } from 'class-validator';
 import { Request, Response } from 'express';
 import {
-  Middleware,
-  ExpressErrorMiddlewareInterface
+  ExpressErrorMiddlewareInterface,
+  Middleware
 } from 'routing-controllers';
-import { DEV_ENV } from '@config';
 import { Service } from 'typedi';
 
 interface ErrorInterface extends Error {
+  name: string;
   httpCode: number;
+  description: string | undefined;
+  errors: string[] | undefined;
 }
 
 @Middleware({ type: 'after' })
 @Service()
 export class ErrorMiddleware implements ExpressErrorMiddlewareInterface {
-  error(
-    error: ErrorInterface,
+  public error(
+    error: any,
```

Comment:
Hi Nico, the problem with that is that not all the errors have the same structure when they arrive to the handler since we have class-validator errors and HttpException errors. 
Therefore I tried but It was not working. 

Anyways good advice.